open-craft/opencraft
# SE-1764 Add field for emails on appserver failure

This is a followup to #478 . It has since been determined that re-using the additional_monitoring_emails field for the purpose of extra alert emails about appserver provision failure is bad, because this field's semantics is usually urgent alerts only.

This adds a new field specifically for extra emails to receive alerts on appserver provision fail.

**Merge deadline**: ASAP this is causing a lot of false positive alerts and blocking other tasks

**Testing instructions**:

1. add an email address to an instance's `provision_failed_emails` and another address to additional monitoring emails.
2. ensure the instance's config is such that the next appserver to launch fails (eg. set `AWS_GATHER_FACTS: true` because that currently fails)
3. launch a new appserver
4. wait until it reaches configuration failed state
5. check the email address' inbox and verify that it received the alert email. also verify that the email in additional_monitoring_emails does *not* receive an alert
6. view the settings for an instance and the configuration for an appserver (from the frontend) and verify the provision failed emails field is correctly displayed
7. launch a new *production* instance (eg. using the prod instance factory from the shell).
8. view the new instance in admin and verify that the emails set in the `PROD_APPSERVER_FAIL_EMAILS` (urgent@ currently on stage) is entered in the `provision_failed_emails` field.

For convenience, this branch is checked out on stage.

**Author notes and concerns**:


**Reviewers**
